keepkeylib/client.py

```diff
@@ -1101,6 +1101,19 @@ def recovery_device(self, use_trezor_method, word_count, passphrase_protection,
         self.init_device()
         return res
 
+    @field('message')
+    @expect(proto.Success)
+    def test_recovery_seed(self, word_count, language):
+        if word_count not in (12, 18, 24):
+            raise Exception("Invalid word count. Use 12/18/24")
+        res = self.call(proto.RecoveryDevice(language=language,
```

Comment:
please pass `word_count` like we have here: https://github.com/keepkey/python-keepkey/pull/86/files#diff-72e2c3d83c723a9f514ee2e1ad8a2dc7R1093

That will make the recovery UX slightly better. Leaving it off will let the device accept 12/18/24 word sentences, but the user has to know to hit return instead of space at the end, which tends to be confusing.

Reply:
I just tested this and that isn't actually how it works, hence why I just left it off in the PR. The user currently needs to press enter/return after the last word, regardless of whether they set how many words we are recovering with. (Pressing space after the last word when setting it to 12 words will throw keepkeylib.client.CallException: (3, u'Too many words entered')

Are you open to changing/improving this behavior in the firmware?

Reply:
> I just tested this yesterday and again today, that isn't actually how it currently works,

During recovery, If you set `word_count` to 12, and hit space after the 12th word, the device tells you that too many words have been entered. If you leave off `word_count`, and hit space after the 12th word, it lets you enter an 18 or 24 word sentence, so long as you hit enter after the 18th or 24th word respectively. Maybe this is different for dry run? I'll have to check.

> Are you open to changing/improving this behavior in the firmware?

Sure, so long as the existing behavior when `word_count == None` is preserved, since I don't want to have to re-engineer how that part of the KeepKey client works. Mind writing up a proposal in an issue against the firmware repo?

Reply:
how about: https://github.com/keepkey/keepkey-firmware/pull/210/files

Were you thinking of something else, improvement-wise?